variables.tf

```
@@ -0,0 +1,14 @@
+variable "ec2_account_settings" {
+  description = "Object of inputs for ec2 account settings"
+  type = object({
+    aws_ebs_encryption_by_default = optional(object({
+      enabled = optional(bool)
+    }))
+    aws_ec2_image_block_public_access = object({
+      state = string
+    })
+    aws_ec2_serial_console_access = optional(object({
+      enabled = bool
+    }))
+  })
+}
```

Streamlining the object naming and setting secure defaults where it makes sense.

```suggestion
variable "ec2_account" {
  description = "Object of inputs for ec2 account settings"
  type = object({
    ebs_encryption_by_default = optional(object({
      enabled         = optional(bool, true)
      default_kms_key = optional(string)
    }), {})
    image_block_public_access = optional(object({
      state = optional(string, "block-new-sharing")
    }), {})
    serial_console_access = optional(object({
      enabled = optional(bool, false)
    }))
  })
  default  = {}
  nullable = false
}
```